src/ensembl/src/content/app/entity-viewer/shared/helpers/entity-helpers.ts

```diff
@@ -191,3 +192,25 @@ export const getProteinXrefs = <
 
   return proteinXrefs;
 };
+
+export const getTranscriptMetadata = (
+  metadata: Pick<TranscriptMetadata, 'canonical' | 'mane'>
+) => {
+  const { canonical, mane } = metadata;
+  if (canonical && mane) {
+    return {
+      label: mane.label,
+      definition: mane.definition
+    };
+  } else if (canonical) {
+    return {
+      label: canonical.label,
+      definition: canonical.definition
+    };
+  } else if (mane) {
+    return {
+      label: mane.label,
+      definition: mane.definition
+    };
+  }
+};
```

Comment:
A very suspicious place for this function. And a very suspicious name for it, in this file (the name made sense in the component file where it was used, but it is confusing here).

Reply:
I agree to that. We already have the metadata when we call this function so there is no point in calling it as `getMetadata`.

I think we could put this back to where it was and then create another function here called `getTranscriptLabelFromMetadata`

Reply:
@manoj Why do you suggest we put it back rather than just rename this function?

Reply:
I renamed the function ✅

Reply:
> Why do you suggest we put it back rather than just rename this function?

This is a function for a very special presentation concern — displaying a transcript label. Is it possibly a React component?

Reply:
@azangru This function is reused in `ensObjectHelpers.ts` as it wants to extract the same thing from metadata: https://github.com/Ensembl/ensembl-client/pull/552/files#diff-620ee37acbb79360a848bc6acc9266eccde25c753f4edb74f70081d1df044eceR161